### ISSUE TITLE
wrong primary key in result for 'hasManyThrough' relation

### ISSUE THREAD

**Issue by Pezhvak** [laravel/framework#27673]
- Laravel Version: 5.7.27
- PHP Version: 7.2.11
- Database Driver & Version: mysql 14.14 Distrib 5.7.24

### Description:
has many through relationship replaces primary key of the related table in the result with localkey

### Steps To Reproduce:
here's the setup:

table Accounts:
 - id
 - pk
 - name

table Relations
- pk
- target_pk

table ProfileInfos
- pk
- col1
- col2
- ..

here's the relation in Account.php model:

```
public function followings(){
        return $this->hasManyThrough(ProfileInfo::class, Relation::class, 'pk', 'pk', 'pk', 'target_pk');
}
```

this returns the data, but pk of table ProfileInfo will get replaces by pk of the account, which is not the desired result. i currently found a work around which is adding `selectRaw('*, instagram_profile_infos.pk as profile_pk')` to the end of relation.

**Comment by staudenmeir** [comment]
This will be fixed in Laravel 5.8: https://github.com/laravel/framework/pull/25812

**Comment by driesvints** [comment]
Please see link above.
